### PR TITLE
[fix] #4084: NumericValue suffix is mandatory

### LINKED ISSUE #4084: [suggestion] `NumericValue` serialization inconsistency
### Feature request

It seems illogical to me that NumericValue should be serialized differently in different genesis instructions
```
{
  "SetKeyValue" : {
    "AssetId" : "foo#wonderland#alice@wonderland",
    "key" : {
      "Name" : "key2"
    },
    "value" : "100"
  }
}
```
```
{
  "Mint" : {
    "object" : "100_u32",
    "destination_id" : {
      "AssetId" : "xor#wonderland#bob@wonderland"
    }
  }
}
```

### Motivation

In Mint instructions it looks fine to me

### Who can help?

@mversic

### PR DESCRIPTION
**Pull request by Erigara**
## Description

Previously suffix for `Fixed` was optional which led to confusion in cases like `{"value": "100"}` which was parsed as `Fixed` nor `String` or integer type. 

After introduced changes `_fx` is required. 

### Linked issue

Closes #4084 

### Benefits

Less confusion when desalinizing.

### CODE REVIEW

**Review by pesterev — APPROVED** [review]

File: data_model/src/numeric.rs [hunk 1]
```diff
@@ -310,7 +306,7 @@ mod tests {
             NumericValue::Fixed((u32::MAX as f64).try_into().expect("trivial conversion")),
         ];
         for val in values {
-            let new_value: NumericValue = format!("{val}").parse().expect("Failed to parse");
+            let new_value: NumericValue = format!("{val:?}").parse().expect("Failed to parse");
```
Comment (DCNick3) [inline data_model/src/numeric.rs:309]: Why did you change this? Does the `Display` implementation not output the suffix?

Comment (Erigara) [reply data_model/src/numeric.rs:309]: Serialization was based on Debug repetition from the beginning, i’m not sure why Display was used in the first place. 
And, yes, afaik display doesn’t print suffixes.

**Review by DCNick3 — DISMISSED** [review]
